core/src/main/scala-3/neotypes/CypherStringInterpolator.scala

```diff
@@ -54,7 +54,14 @@ object CypherStringInterpolator {
           '{ (b: Boolean) =>
             if (b)
               $qarg
-            else Left($expr.toString)
+            else
+              // This `asInstanceOf` is necessary because when some class overrides `toString()` without parentheses,
+              // this line causes compile error during macro expansion. Scala distinguishes `toString` from `toString()`,
+              // the former is `Select(tree,"toString")` and the latter is `Apply(tree, "toString", typeArgs = Nil, args = Nil)`.
+              //
+              // For example, `cats.data.Chain` overrides to string `override def toString: String = show(Show.fromToString)`.
+              // Without this `asInstanceOf`, the test for Chain in BaseCatsDataSpec won't compile.
+              Left($expr.asInstanceOf[Object].toString())
```

Comment:
Wonderfull @.@

Reply:
While minimizing this case, I found a solution that avoids asInstanceOf. It is late at night in my timezone, so I'll work on it tommorow or later.

 
https://discord.com/channels/632150470000902164/875868146949554207/1146122590037999726

Reply:
done